src/pubd/manager.rs

```diff
@@ -604,18 +597,44 @@ mod tests {
     }
 
     fn session_dir_contains_delta(session_path: &Path, serial: u64) -> bool {
-        let mut path = session_path.to_path_buf();
-        path.push(format!("{}/delta.xml", serial));
-        path.exists()
+        find_in_serial_dir(session_path, serial, "delta.xml").is_some()
     }
 
     fn session_dir_contains_snapshot(session_path: &Path, serial: u64) -> bool {
-        session_dir_snapshot(session_path, serial).exists()
+        find_in_serial_dir(session_path, serial, "snapshot.xml").is_some()
     }
 
-    fn session_dir_snapshot(session_path: &Path, serial: u64) -> PathBuf {
-        let mut path = session_path.to_path_buf();
-        path.push(format!("{}/snapshot.xml", serial));
-        path
+    fn session_dir_snapshot(session_path: &Path, serial: u64) -> Option<PathBuf> {
+        find_in_serial_dir(session_path, serial, "snapshot.xml")
+    }
+
+    /// Expects files (like delta.xml or snapshot.xml) under dir structure like:
+    /// <session_path>/<serial>/<some random>/<filename>
+    fn find_in_serial_dir(session_path: &Path, serial: u64, filename: &str) -> Option<PathBuf> {
+        let serial_dir = session_path.join(serial.to_string());
+        if let Ok(randoms) = fs::read_dir(&serial_dir) {
+            for entry in randoms {
+                let entry = entry.unwrap();
```

Comment:
This was safe, because it's done in test code only. However.. I have now moved this code to `RrdpServer` so that it could be reused to clean up out of scope snapshots and made it more careful.